### PR TITLE
[ADD] allow to connect SFTP using a ssh private key

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**

### CODE REVIEW

**Comment by Cedric-Pigeon** [comment]
@sebastienbeau

File: storage_backend_sftp/components/sftp_adapter.py [hunk 1]
```diff
@@ -35,7 +36,12 @@ def sftp(backend):
     account = backend._get_keychain_account()
     password = account._get_password()
     transport = paramiko.Transport((backend.sftp_server, backend.sftp_port))
-    transport.connect(username=backend.sftp_login, password=password)
+    if backend.sftp_auth_method == 'pwd':
+        transport.connect(username=backend.sftp_login, password=password)
+    elif backend.sftp_auth_method == 'ssh_key':
+        ssh_key_buffer = StringIO(password)
+        private_key = paramiko.RSAKey.from_private_key(ssh_key_buffer)
+        transport.connect(username="akeneo", pkey=private_key)
```
Comment (andreparames) [inline storage_backend_sftp/components/sftp_adapter.py]: Shouldn't the username be `backend.sftp_login`?

File: storage_backend_sftp/components/sftp_adapter.py [hunk 2]
```diff
@@ -35,7 +36,12 @@ def sftp(backend):
     account = backend._get_keychain_account()
     password = account._get_password()
     transport = paramiko.Transport((backend.sftp_server, backend.sftp_port))
-    transport.connect(username=backend.sftp_login, password=password)
+    if backend.sftp_auth_method == 'pwd':
+        transport.connect(username=backend.sftp_login, password=password)
+    elif backend.sftp_auth_method == 'ssh_key':
+        ssh_key_buffer = StringIO(password)
+        private_key = paramiko.RSAKey.from_private_key(ssh_key_buffer)
```
Comment (andreparames) [inline storage_backend_sftp/components/sftp_adapter.py]: While RSA is the default, both ssh and paramiko support other formats, such as ECDSA. In the paramiko client class, they just try each in sucession until one works, we could do the same:

```python
def load_ssh_key(ssh_key_buffer):
    for pkey_class in (RSAKey, DSSKey, ECDSAKey, Ed25519Key):
        try:
            return pkey_class.from_private_key(ssh_key_buffer)
        except SSHException:
            ssh_key_buffer.seek(0) # reset the buffer "file"
    raise Exception("Invalid ssh private key")

**Comment by Cedric-Pigeon** [comment]
@andreparames Thanks, I included your suggestion.

**Review by andreparames — APPROVED** [review]

**Pull request merged**